File: python/Petshop.py
class Petshop:
    NextID = 1  # variabel statis untuk autoincrement id
    _ID = NextID
    _produk = ""
    _harga = 0
    _stok = 0

    #constructor
    def __init__(self, produk, harga, stok):
        self._ID = Petshop.NextID
        Petshop.NextID += 1
        self._produk = produk
        self._harga = harga
        self._stok = stok

    #getter
    def get_ID(self):
        return self._ID
    def get_produk(self):
        return self._produk
    def get_harga(self):
        return self._harga
    def get_stok(self):
        return self._stok
    
    #setter
    def set_produk(self, produk):
        self._produk = produk
    def set_harga(self, harga):
        self._harga = harga
    def set_stok(self, stok):
        self._stok = stok

File: php/Petshop.php
<?php

class Petshop {
    private static $NextID = 1; // Variabel statis untuk autoincrement id
    private $ID;
    protected $produk;
    protected $harga;
    protected $stok;
    protected $foto;

    // construct dengan parameter
    public function __construct($produk, $harga, $stok, $foto) {
        self::$NextID++; // self fungsinya supaya ketika instansiasi objek tidak mengenali variabel ini sebegai variabel yang sama dengan objek yang diinstansiasi sebelumnya (jadi kayak global var gitu) 
        $this->ID = self::$NextID; // self:: (variabel statis, sama seperti static di c++ dan java)
        $this->produk = $produk;
        $this->harga = $harga;
        $this->stok = $stok;
        $this->foto = $foto;
    }

    public function getID() {
        return $this->ID;
    }
    public function getProduk() {
        return $this->produk;
    }
    public function getHarga() {
        return $this->harga;
    }
    public function getStok() {
        return $this->stok;
    }
    public function getFoto() {
        return $this->foto;
    }


    public function setID($id) {
        $this->ID = $id;
    }
    public function setProduk($produk) {
        $this->produk = $produk;
    }
    public function setHarga($harga) {
        $this->harga = $harga;
    }
    public function setStok($stok) {
        $this->stok = $stok;
    }
    public function setFoto($foto) {
        $this->foto = $foto;
    }
}